### DISCUSSION TITLE
Migration Script from Sqlite to Etcd Datastore

### DISCUSSION THREAD

**Discussion by ujala-singh** [k3s-io/k3s#9777] [Ideas]
Hi Team,

I am kinda looking for a solution where I want to migrate from SQLite to etcd data store. Can someone help me with the script or open-source code?

I can't follow this https://docs.k3s.io/datastore/ha-embedded#existing-single-node-clusters, I have some environment constraints.

Use Case is, I am running a vCluster within a k8s cluster which uses k3s distro with embedded SQLite by default, but there are some constraints involved where I can not directly run ```--cluster-init```  flag and migrate to embedded etcd directly. Hence I require some help to migrate it from scratch.

**Comment by brandond** [comment]
vcluster does not provide a way to add the `--cluster-init` flag to the k3s command-line arguments? This sounds like something to take up with the vcluster project maintainers.
